### PR TITLE
Improve board fvp_baser_aemv8r documentation

### PR DESCRIPTION
**Pull request by fengqi-w**
Improve board fvp_baser_aemv8r documentation:
1. Add a new page `Debug with Arm DS` to describe how to debug this board with Arm Development Studio.
2. Standardize the names used in the document.

For commit `boards: arm64: doc: list files in toctree explicitly`, if glob are used in toctree, the file `debug-with-arm-ds` will appear in the board level table of contents. See the following picture for details. Also see thread https://zephyrproject.slack.com/archives/CPY1UNDLK/p1618987049043500 for more discussions.

toc level is wrong

### CODE REVIEW

**Comment by carlocaione** [comment]
From the screenshot it looks like the menu is under `ARM64 Boards` and not under `Arm FVP ...` where I was expecting to find it. Any reason for this?

**Comment by fengqi-w** [comment]
The screenshot is to trying to show the problem that commit `boards: arm64: doc: list files in toctree explicitly` is trying to solve. Not what it will look like after this PR. Sorry if it's confusing.

The following picture shows what it looks like if this PR is merged:
toc-correct

**Comment by carlocaione** [comment]
> The screenshot is to trying to show the problem that commit `boards: arm64: doc: list files in toctree explicitly` is trying to solve. Not what it will look like after this PR. Sorry if it's confusing.

Ohh, ok, then. Maybe I need more coffee this morning :)

File: boards/arm64/index.rst [hunk 1]
```diff
@@ -5,6 +5,9 @@ ARM64 Boards
...
 .. toctree::
    :maxdepth: 1
-   :glob:
 
-   **/*
+   bcm958402m2_a72/doc/index.rst
+   fvp_baser_aemv8r/doc/index.rst
+   nxp_ls1046ardb/doc/index.rst
+   qemu_cortex_a53/doc/index.rst
+   xenvm/doc/index.rst
```
Comment (carlocaione) [inline boards/arm64/index.rst:13]: I'm not a documentation expert but I wonder if there is a way to avoid listing all the index files here.

Comment (fengqi-w) [reply boards/arm64/index.rst:13]: I tried something like `*/doc/index.rst`, but it doesn't work, gives me this error:
```
Warning, treated as error:
/workdir/zephyr/doc/_build/rst/doc/boards/arm64/index.rst:6:toctree glob pattern '*/doc/index.rst' didn't match any documents
ninja: build stopped: subcommand failed.
```
I'm also curious if there's a better way.

Comment (fengqi-w) [reply boards/arm64/index.rst:13]: `**/index.rst` is not working, got similar error:
```
Warning, treated as error:
/workdir/zephyr/doc/_build/rst/doc/boards/arm64/index.rst:6:toctree glob pattern '**/index.rst' didn't match any documents
ninja: build stopped: subcommand failed.
```

I also had tried some other patterns like `**index.rst`, `**/doc/index.rst`, `**/*.rst`, and
```
*/
*/doc/
*/doc/index.rst
```
and
```
*/doc/index.rst
*/doc
*
```
and some other patterns I've tried days ago, which I can't remember now.

Comment (gmarull) [reply boards/arm64/index.rst:13]: I have done a quick try and this should work:

```rst
.. toctree::
    :glob:

    **/index
```

Comment (fengqi-w) [reply boards/arm64/index.rst:13]: It does work! I'll update the patch soon.

**Review by gmarull — APPROVED** [review]
doc changes LGTM, I'll leave content review to experts on this topic

**Review by carlocaione — APPROVED** [review]
LGTM

**Pull request merged**